tests/scripts/actionutils.sh

```diff
@@ -824,6 +851,198 @@ function bombard_rgw_configs() {
     sudo microceph.ceph health
 }
 
+
+# Check if the cluster has osd noout flag set.
+# Usage: is_osd_noout_set
+function is_osd_noout_set() {
+    set -xe
+
+    microceph.ceph osd dump | grep noout > /dev/null 2>&1
+}
+
+# Check if the snap service is active and enabled
+# Usage: check_snap_service_active_enabled <microceph service name>
+function check_snap_service_active_enabled() {
+    local service="${1?missing}"
+
+    set -xe
+
+    snap services microceph.$service | grep active > /dev/null 2>&1
+    snap services microceph.$service | grep enabled > /dev/null 2>&1
+}
+
+# Test dry run `microceph cluster maintenance enter` prints expected number of steps.
+# Usage: test_dry_run_maintenance_enter <node name>
+function test_dry_run_maintenance_enter() {
+    local node="${1?missing}"
+
+    set -xe
+
+    # Debug
+    nodeexec $node "microceph status"
+    nodeexec $node "microceph.ceph -s"
+
+    # Count expected steps when --set-noout=false --stop-osds=false
+    lines=$(nodeexec $node "microceph cluster maintenance enter $node --set-noout=false --stop-osds=false --dry-run | tee output.txt | wc -l")
+    nodeexec $node "cat output.txt"
+    if [ $lines != "3" ]; then exit 1; fi
```

Comment:
It would be great if the test func would log some additional details on the failure to make the issue more obvious / easier to debug. Could e.g. log the contents of the output.txt

Similarly in the tests below.

Reply:
Added